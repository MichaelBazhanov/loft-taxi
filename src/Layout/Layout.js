import React from "react";
import { Outlet } from "react-router-dom";
import Header from "../components/Header";
import PropTypes from "prop-types";
import { useLocation } from "react-router-dom";
import Notification from "../components/Notification";
import { connect } from "react-redux";
import { hideNotification } from "../modules/tooltips";

const Layout = ({
  isTooltipShown,
  tooltipText,
  tooltipType,
  hideNotification,
}) => {
  const location = useLocation();
  const showedHeader =
    location.pathname !== "/" &&
    location.pathname !== "/login" &&
    location.pathname !== "/registration";

  return (
    <div className="App sans antialiased grid auto-cols-fr grid-rows-[auto_1fr] relative h-screen">
      {showedHeader && <Header />}

      <main>
        <section className="h-full bg-black-me">
          <Outlet />
        </section>
      </main>

      <div className="fixed top-full left-1/2 -translate-x-1/2">
        <div
          className={
            isTooltipShown
              ? "transition-transform duration-300 -translate-y-full"
              : "transition-transform duration-300"
          }
        >
          <Notification
            text={tooltipText}
            type={tooltipType}
            onClick={() => hideNotification()}
          />
        </div>
      </div>
    </div>
  );
};

Layout.propTypes = {
  hideNotification: PropTypes.func,
  isTooltipShown: PropTypes.bool,
  tooltipText: PropTypes.string,
  tooltipType: PropTypes.string,
};

export default connect(
  (state) => ({
    isTooltipShown: state.tooltipsReducer.isShown,
    tooltipText: state.tooltipsReducer.text,
    tooltipType: state.tooltipsReducer.type,
  }),
  { hideNotification }
)(Layout);
