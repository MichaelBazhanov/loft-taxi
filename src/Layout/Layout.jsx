import React from "react";
import { Outlet } from "react-router-dom";

import Header from '../components/Header'

import { useLocation } from 'react-router-dom'

const Layout = () => {
	let location = useLocation();

	return (
		<div className='App sans antialiased'>
			{(location.pathname !== '/login' && location.pathname !== '/registration') && <Header />}
			{/* <Header /> */}
			<main>
				<section className="bg-black-me">
					<Outlet />
				</section>
			</main>
		</div >
	)
}

export default Layout

