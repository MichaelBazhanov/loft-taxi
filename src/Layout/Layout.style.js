export default {
  container: [
    "font-sans",
    "antialiased",
    "grid",
    "auto-cols-fr",
    "grid-rows-[auto_1fr]",
    "relative",
    "h-screen",
  ].join(" "),
  section: ["h-full", "bg-black-me"].join(" "),
  "container-notification": [
    "fixed",
    "top-full",
    "left-1/2",
    "-translate-x-1/2",
  ].join(" "),
  "tooltip-show": ({ isTooltipShown }) =>
    [
      "transition-transform",
      "duration-300",
      isTooltipShown ? "-translate-y-full" : "",
    ].join(" "),
};
