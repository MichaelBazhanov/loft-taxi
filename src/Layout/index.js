export { default } from './Layout'
export * from './selectors'
export * from './styles'