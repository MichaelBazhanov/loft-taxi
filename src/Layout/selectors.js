import { createSelector } from "reselect";

export const getTooltipsIsShow = createSelector(
  [(state) => state.tooltipsReducer.isShown],
  (isShown) => isShown
);
export const getTooltipsText = createSelector(
  [(state) => state.tooltipsReducer.text],
  (text) => text
);
export const getTooltipsType = createSelector(
  [(state) => state.tooltipsReducer.type],
  (type) => type
);
