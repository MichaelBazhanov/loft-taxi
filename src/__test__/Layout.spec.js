import React from 'react'
import { Route, Routes } from 'react-router-dom'
import { render } from '@testing-library/react'
import TestRenderer from 'react-test-renderer'

import Layout from '../Layout/Layout'

it('Renders correctly', () => {
  const layout = TestRenderer.create(<Layout />).toJSON() // превращаем компонент в JSON

  render(
    <Routes>
      <layout />
    </Routes>,
  )

  expect(layout).toMatchSnapshot()
})

