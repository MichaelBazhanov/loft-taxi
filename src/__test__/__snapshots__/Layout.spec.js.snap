// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="font-sans antialiased grid auto-cols-fr grid-rows-[auto_1fr] relative h-screen"
>
  <main>
    <section
      className="h-full bg-black-me"
    />
  </main>
  <div
    className="fixed top-full left-1/2 -translate-x-1/2"
  >
    <div
      className="transition-transform duration-300 "
    >
      <div
        className="min-w-[400px] relative flex justify-between items-center text-white rounded-t-lg success"
      >
        <p
          className="pl-5 select-none"
        >
          
        </p>
        <button
          className="p-5 text-white ml-5 cursor-pointer hover:text-white transition-colors duration-500 ease-in-out"
          onClick={[Function]}
        >
          <svg
            className="h-5 w-5 stroke-current stroke-[3px]"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
